### ISSUE TITLE
Upgrade `cobra` version to use `completion` command

### ISSUE THREAD

**Issue by dhruvmanila** [deepsourcelabs/cli#170]
Current version: `1.1.3`
Command `completion` introduced in: [`1.2.0`](https://github.com/spf13/cobra/releases/tag/v1.2.0)

This can then be used to generate completions automatically as per the user shell. It can be integrated in Homebrew formulae as well.

**Comment by burntcarrot** [comment]
Thanks for the heads up! We're actively working on upgrading the versions for our dependencies; the upcoming releases should have support for code completions.

**Comment by siddhant-deepsource** [comment]
Upgrading to the latest versions of `cobra` will also help us to remove `viper` as a dependency. Ref: https://github.com/spf13/cobra/releases/tag/v1.4.0

@dhruvmanila While we shall anyways be upgrading it in upcoming release but if you want to send a PR to fix this, feel free to. Thanks 🙌

**Comment by dhruvmanila** [comment]
@siddhant-deepsource Sure, I was going to. Should I be including the changes like updating the `configvalidator` package, using `completion` command, etc. in the same PR or keep them separate? I'm not really sure what you guys follow, so thought to ask :)

**Comment by siddhant-deepsource** [comment]
Yep, let's keep them in separate PRs. We usually ensure that one PR addresses one concern.
